File: 03_Manipulating_Items/03_updateItem.js

// SET is upsert
var params = {
    TableName: 'productTable',
    Key: {

        "id": 20,
        "type": "book"
    },
    UpdateExpression: 'SET stock = :value, title = :titlevalue', 
    ExpressionAttributeValues: { 
        ':value': 97,
        ':titlevalue' : "book 1",
        ':num' : 98
    },
    ConditionExpression: "stock >= :num",
    ReturnValues:"ALL_NEW"
};
docClient.update(params, function(err, data) {
    if (err) ppJson(err); // an error occurred
    else ppJson(data); // successful response
});

File: 03_Manipulating_Items/03_UpdateItem.cs
AmazonDynamoDBClient client = new AmazonDynamoDBClient();
string tableName = "productTable";

var request = new UpdateItemRequest
{
    TableName = tableName,
    Key = new Dictionary<string,AttributeValue>() { { "id", new AttributeValue { N = "20" } },
    { "type", new AttributeValue { S = "phone" }},
    ExpressionAttributeNames = new Dictionary<string,string>()
    {
        {"#R", "rating"},
        {"#S", "stock"},
        {"#M", "mnfr"}
    },
    ExpressionAttributeValues = new Dictionary<string, AttributeValue>()
    {
        {":val1",new AttributeValue { N = "5"}},
        {":val2",new AttributeValue {N = "2"}}
    },
    UpdateExpression = "ADD #R :val1 SET #S = #S - :val2 REMOVE #M"
};

var response = client.UpdateItem(request);

//conditional update example
var request = new UpdateItemRequest
{
    Key = new Dictionary<string,AttributeValue>() { { "id", new AttributeValue { N = "10" } }, { "type", new AttributeValue { S = "phone" } }    }, ExpressionAttributeNames = new Dictionary<string,string>()
    {
        {"#S", "stock"}
    },
    ExpressionAttributeValues = new Dictionary<string, AttributeValue>()
    {
        {":newprice",new AttributeValue {N = "50"}},// Update stock to 50 only if
        {":currprice",new AttributeValue {N = "20"}} // Current value is 20
    },
    UpdateExpression = "SET #S = :newprice",
    ConditionExpression = "#S = :currprice",
    TableName = tableName,
    ReturnValues = "ALL_NEW" 
};